src/onnx/include/migraphx/onnx/onnx_parser.hpp

```diff
@@ -58,25 +59,30 @@ struct onnx_parser
     using op_func  = std::function<std::vector<instruction_ref>(
         onnx_parser&, const node_info&, std::vector<instruction_ref>)>;
     node_map nodes;
-    std::unordered_map<std::string, instruction_ref> instructions;
     program prog                  = program();
     std::size_t default_dim_value = 1;
     std::unordered_map<std::string, std::vector<std::size_t>> map_input_dims;
     bool skip_unknown_operators = false;
+    int64_t max_iter_num        = 10;
     int64_t opset_version       = 13;
 
     std::unordered_map<std::string, op_func> ops;
 
     onnx_parser();
     operation load(const std::string& name, const node_info& info) const;
 
-    void parse_undefined(module* mod, const std::string& name);
+    void parse_undefined(module* mod,
+                         const std::string& name,
+                         std::unordered_map<std::string, instruction_ref>& instructions);
 
     static int64_t get_opset_version(const onnx::ModelProto& model);
 
     void parse_from(std::istream& is, std::string name = "");
     void parse_from(const void* data, std::size_t size);
-    void parse_graph(module* mod, const onnx::GraphProto& graph);
+    void parse_graph(module* mod,
+                     const onnx::GraphProto& graph,
+                     std::unordered_map<std::string, instruction_ref> instructions = {},
```

Comment:
Why is this passed as a parameter instead of using the member variable?

Reply:
In this way, subgraph can access variables in parent graph but not in the other way.

Reply:
That would be an invalid onnx file. I would prefer to keep it as a member variable. If we want to make sure we dont reference it by accident then we could remove the instructions in the map when `parse_graph` finishes.

Reply:
I still think this would be good to keep instructions as a member variable.

Reply:
fixed